aggsender/db/aggsender_db_storage_test.go

```diff
@@ -170,32 +197,47 @@ func Test_Storage(t *testing.T) {
 
 		// Test fetching certificates with status Settled
 		statuses := []agglayertypes.CertificateStatus{agglayertypes.Settled}
-		certificatesFromDB, err := storage.GetCertificatesByStatus(statuses)
+		certificatesFromDB, err := storage.GetCertificatesByStatus(statuses, false)
 		require.NoError(t, err)
 		require.Len(t, certificatesFromDB, 1)
 		require.ElementsMatch(t, []*types.CertificateInfo{certificates[0]}, certificatesFromDB)
 
 		// Test fetching certificates with status Pending
 		statuses = []agglayertypes.CertificateStatus{agglayertypes.Pending}
-		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses)
+		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses, false)
 		require.NoError(t, err)
 		require.Len(t, certificatesFromDB, 1)
 		require.ElementsMatch(t, []*types.CertificateInfo{certificates[1]}, certificatesFromDB)
 
 		// Test fetching certificates with status InError
 		statuses = []agglayertypes.CertificateStatus{agglayertypes.InError}
-		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses)
+		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses, false)
 		require.NoError(t, err)
 		require.Len(t, certificatesFromDB, 1)
 		require.ElementsMatch(t, []*types.CertificateInfo{certificates[2]}, certificatesFromDB)
 
 		// Test fetching certificates with status InError and Pending
 		statuses = []agglayertypes.CertificateStatus{agglayertypes.InError, agglayertypes.Pending}
-		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses)
+		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses, false)
 		require.NoError(t, err)
 		require.Len(t, certificatesFromDB, 2)
 		require.ElementsMatch(t, []*types.CertificateInfo{certificates[1], certificates[2]}, certificatesFromDB)
 
+		// Test fetching certificates with status InError, but with light query
+		statuses = []agglayertypes.CertificateStatus{agglayertypes.InError}
+		certificatesFromDB, err = storage.GetCertificatesByStatus(statuses, true)
+		require.NoError(t, err)
+		require.Len(t, certificatesFromDB, 1)
+		require.Equal(t, certificates[2].Height, certificatesFromDB[0].Height)
+		require.Equal(t, certificates[2].RetryCount, certificatesFromDB[0].RetryCount)
+		require.Equal(t, certificates[2].CertificateID, certificatesFromDB[0].CertificateID)
+		require.Equal(t, certificates[2].Status, certificatesFromDB[0].Status)
+		require.Equal(t, certificates[2].CreatedAt, certificatesFromDB[0].CreatedAt)
+		require.Equal(t, certificates[2].UpdatedAt, certificatesFromDB[0].UpdatedAt)
+		require.Equal(t, certificates[2].FromBlock, certificatesFromDB[0].FromBlock)
+		require.Equal(t, certificates[2].ToBlock, certificatesFromDB[0].ToBlock)
```

Comment:
Hm, why this wouldn't work? It is tedious to write all those assertions...
```suggestion
		require.Equal(t, certificates[2], certificatesFromDB[0])
```

Reply:
Its because they should be the same except for the `AggchainProof` field, which should be nil in the case of the `certficatesFromDB[0]`, because we used the light query option, in which we do not load the proof from `db`.